tutorials/fastapi-agent/scripts/fastapi_agent.py

```
@@ -1,91 +1,183 @@
+    """Process a batch of queries and return all responses."""
+    results = []
+    for req in requests:
+        try:
+            cached = get_cached_response(req.query, req.context)
+            if cached:
+                results.append({"query": req.query, "response": cached, "cached": True})
+                continue
+            response = agent.generate_response(req.query)
+            set_cached_response(req.query, response, req.context)
+            results.append({"query": req.query, "response": response, "cached": False})
+        except:
```

Ruff (0.15.17)

[error] Do not use bare `except`

(E722)